core/src/main/java/tc/oc/pgm/filters/dynamic/FilterMatchModule.java

```diff
@@ -107,9 +141,21 @@ public void onMatchLoad(MatchLoadEvent event) {
             });
   }
 
+  private void findAndCreateReactorFactories(Filter filter) {
+    if (filter instanceof ParentFilter) {
+      ((ParentFilter) filter).getChildren().forEach(this::findAndCreateReactorFactories);
+    }
+    if (filter instanceof ReactorFactory) {
+      final ReactorFactory.Reactor reactor =
+          ((ReactorFactory<?>) filter).createReactor(this.match, this);
+      this.activeReactors.put((ReactorFactory<? extends ReactorFactory.Reactor>) filter, reactor);
```

Comment:
You should check that a reactor doesn't already exist before creating another. Currently you could be potentially creating it twice if someone did this xml:

```
<filters>
  <time id="after-5min">5m</time>
  <all>
    <filter id="after-5min"/>
  </all>
</filters>
```
First after-5min would get registered for being ReactorFactory, then <all> would get registered for being a ParentFilter, creating the reactor once again.

Also, are references resolved? if a filter references another it probably should be the referenced one which is checked for instance of reactor factory.

Reply:
References should all be resolved at this time, so ill add a case for it. Good catch